### PR TITLE
client: Add container template files operations

### PR DESCRIPTION
**Pull request by albertodonato**
Add a new `lxc config template <op>` section to the CLI to perform operations related to template files.

### CODE REVIEW

File: doc/rest-api.md [hunk 1]
```diff
@@ -1268,7 +1268,7 @@ Return:
 * Operation: Sync
 * Return: the content of the template
 
-## POST (?path=\<template\>)
+### POST (?path=\<template\>)
```
Comment (brauner) [inline doc/rest-api.md]: Just to make sure: Is this intentional? :)

File: client/lxd_containers.go [hunk 2]
```diff
@@ -1304,3 +1305,102 @@ func (r *ProtocolLXD) SetContainerMetadata(name string, metadata api.ImageMetada
 
 	return nil
 }
+
+// GetContainerTemplateFiles returns the list of name of a template files for a container.
```
Comment (brauner) [inline client/lxd_containers.go]: s/GetContainerTemplateFiles returns the list of name of a template files for a container./GetContainerTemplateFiles returns the list of names of template files for a container.
?

File: client/lxd_containers.go [hunk 3]
```diff
@@ -1304,3 +1305,102 @@ func (r *ProtocolLXD) SetContainerMetadata(name string, metadata api.ImageMetada
...
+func (r *ProtocolLXD) GetContainerTemplateFiles(containerName string) ([]string, error) {
+	if !r.HasExtension("container_edit_metadata") {
+		return nil, fmt.Errorf("The server is missing the required \"container_edit_metadata\" API extension")
+	}
+
+	templates := []string{}
+
+	url := fmt.Sprintf("/containers/%s/metadata/templates", containerName)
```
Comment (brauner) [inline client/lxd_containers.go]: Does this need an `api.Version` prefix?

File: client/lxd_containers.go [hunk 4]
```diff
@@ -1304,3 +1305,102 @@ func (r *ProtocolLXD) SetContainerMetadata(name string, metadata api.ImageMetada
...
+
+	return resp.Body, err
+}
+
+// CreateContainerTemplateFile creates an empty file template for a container.
+func (r *ProtocolLXD) CreateContainerTemplateFile(containerName string, templateName string) error {
+	return r.setContainerTemplateFile(containerName, templateName, bytes.NewReader(nil), "POST")
+}
+
+// UpdateContainerTemplateFile updates content for a container template file.
```
Comment (brauner) [inline client/lxd_containers.go]: s/updates content/updates the content/

File: lxc/config.go [hunk 5]
```diff
@@ -1246,3 +1348,53 @@ func (c *configCmd) doContainerMetadataEdit(client lxd.ContainerServer, name str
...
+		return err
+	}
+
+	for {
+		reader := bytes.NewReader(content)
+		err := client.UpdateContainerTemplateFile(containerName, templateName, reader)
+		// Respawn the editor
+		if err != nil {
+			fmt.Fprintf(os.Stderr, i18n.G("Error updating template file: %s")+"\n", err)
+			fmt.Println(i18n.G("Press enter to start the editor again"))
```
Comment (brauner) [inline lxc/config.go]: Is this the same error message we give in all of or other client tools?

File: doc/rest-api.md (hunk 1 above)
Comment (albertodonato) [inline doc/rest-api.md]: It is, to match other subsections

File: lxc/config.go (hunk 5 above)
Comment (albertodonato) [inline lxc/config.go]: are you refering to the "Press enter..." section?

File: client/lxd_containers.go (hunk 2 above)
Comment (albertodonato) [inline client/lxd_containers.go]: Bad copy/paste/edit. fixed

File: client/lxd_containers.go (hunk 3 above)
Comment (albertodonato) [inline client/lxd_containers.go]: No, .query() does that automatically

File: client/lxd_containers.go (hunk 4 above)
Comment (albertodonato) [inline client/lxd_containers.go]: Fixed

File: lxc/config.go (hunk 5 above)
Comment (brauner) [inline lxc/config.go]: Yep.

Comment (albertodonato) [inline lxc/config.go]: Then yes, It is, in cases where the editor is spawned (text is not read from stdin)

Comment (brauner) [inline lxc/config.go]: cool :)

File: client/interfaces.go [hunk 6]
```diff
@@ -89,6 +89,12 @@ type ContainerServer interface {
 	GetContainerMetadata(name string) (*api.ImageMetadata, string, error)
 	SetContainerMetadata(name string, metadata api.ImageMetadata, ETag string) error
 
+	GetContainerTemplateFiles(containerName string) (templates []string, err error)
+	GetContainerTemplateFile(containerName string, templateName string) (content io.ReadCloser, err error)
+	CreateContainerTemplateFile(containerName string, templateName string) (err error)
+	UpdateContainerTemplateFile(containerName string, templateName string, content io.ReadSeeker) (err error)
```
Comment (stgraber) [inline client/interfaces.go]: Feels to me like Create and Update should both take the same arguments. If someone wants to create an existing file or empty an existing one, they can pass an empty ReadSeeker.

**Comment by albertodonato** [comment]
@stgraber I changed the CreateContainerTemplateFile to match the API. I'm not sure if we want/need to allow passing content in the `create` command too in the CLI.

**Pull request merged**